addons/mail/static/src/new/core/messaging.js

```diff
@@ -380,6 +381,9 @@ export class Messaging {
                         }
                     }
                     this.state.discuss.inbox.counter = needaction_inbox_counter;
+                    if (this.state.discuss.inbox.counter === 0) {
+                        this.showRainbowMan();
+                    }
```

Comment:
This is not the correct place to show rainbow man.
Here being in messaging and in response to `mark as read`, this will be spammed all the time when receiving this notification and inbox has counter 0, regardless of current screen of user.

The intended visual is to show it only once, when inbox counter change to non-zero to zero, and only with discuss app shown. So the `useEffect` should be in the discuss component itself.

Reply:
I don't get it exactly. You mean there are other situations that reduces the counter of inbox but we don't have `mark as read` for them?
I tried `useEffect`  and when I saw it is happening here, I changed it to `useService('effect')`.